aptos-move/framework/aptos-framework/sources/Table.move

```diff
@@ -1,152 +1,108 @@
-/// This module provides a temporary solution for tables by providing a layer on top of Vector
+/// Type of large-scale storage tables.
 module AptosFramework::Table {
     use Std::Errors;
-    use Std::Option;
-    use Std::Vector;
 
-    const EKEY_ALREADY_EXISTS: u64 = 0;
-    const EKEY_NOT_FOUND: u64 = 1;
+    // native code raises this with Errors::invalid_arguments()
+    const EALREADY_EXISTS: u64 = 100;
+    // native code raises this with Errors::invalid_arguments()
+    const ENOT_FOUND: u64 = 101;
+    const ENOT_EMPTY: u64 = 102;
 
-    struct Table<Key: store, Value: store> has store {
-      data: vector<TableElement<Key, Value>>,
+    /// Type of tables
+    struct Table<phantom K, phantom V> has store {
+        handle: u128,
+        length: u64,
     }
 
-    struct TableElement<Key: store, Value: store> has store {
-        key: Key,
-        value: Value,
-    }
-
-    public fun count<Key: store, Value: store>(table: &Table<Key, Value>): u64 {
-        Vector::length(&table.data)
-    }
-
-    public fun create<Key: store, Value: store>(): Table<Key, Value> {
-        Table {
-            data: Vector::empty(),
+    /// Create a new Table.
+    public fun new<K, V: store>(): Table<K, V> {
+        Table{
+            handle: new_table_handle(),
```

Comment:
How do we ensure the uniqueness of this handle across tables?

Reply:
This is the implementation: https://github.com/move-language/move/blob/be45aa0cc24f8b7106979bdf3313c70eb8d16a12/language/extensions/move-table-extension/src/lib.rs#L336-L362

Handles are based on the txn id and the sequence number among the tables being created in the same table.